app/callbacks/views/sns.py

```diff
@@ -428,9 +435,26 @@ def _handle_s3_sns_record(record, message_id):
             # TODO?         note the impossibility of doing this without some race conditions
 
             notify_client = DMNotifyClient(current_app.config["DM_NOTIFY_API_KEY"])
+
+            if (
+                len(clamd_result) >= 2 and
+                clamd_result[1].lower() == current_app.config["DM_EICAR_TEST_SIGNATURE_RESULT_STRING"].lower()
+            ):
+                notify_kwargs = {
+                    # we'll use the s3 ETag of the file as the notify ref - it will be the only piece of information
+                    # that will be shared knowledge between a functional test and the application yet also allow the
+                    # test to differentiate the results of its different test runs, allowing it to easily check for
+                    # the message being sent
```

Comment:
:tophat: :rabbit2: There's no magic wand emoji.

Reply:
In short - it doesn't exist yet, but it will be generating a random "file" containing eicar which it will upload to the app (where? don't know yet). It will then poll the notify "sent" list for a reference matching the md5sum of the randomized file until it finds it. Up to a certain time limit.

Reply:
OK. Would it be useful to include the stage in the reference too,  if say, we are running func tests on staging and preview test run at the same time and want to see which one has appeared in Notify?

Reply:
We *could*, though the possibility of the two randomized files colliding would be insanely unlikely.

Reply:
I was thinking more for debugging purposes - if there was a problem somewhere we'd know where to start looking.

Reply:
Oooooohok.